Source/WebCore/rendering/cocoa/RenderThemeCocoa.mm

```diff
@@ -3744,6 +3749,35 @@ static inline bool shouldAdjustTextControlInnerElementStyles(const RenderStyle&
 
     return { 0, 0 };
 }
+
+float RenderThemeCocoa::adjustedMaximumLogicalWidthForControl(const RenderStyle& style, const Element& element, float maximumLogicalWidth) const
+{
+#if PLATFORM(MAC)
+    if (!style.hasUsedAppearance() || style.nativeAppearanceDisabled())
+        return maximumLogicalWidth;
+
+    const auto isInlineFlipped = style.writingMode().isInlineFlipped();
+    const auto isHorizontal = style.writingMode().isHorizontal();
+
+    const auto inlineEndPaddingExplicitlySet = isHorizontal ? (isInlineFlipped ? style.hasExplicitlySetPaddingLeft() : style.hasExplicitlySetPaddingRight()) : (isInlineFlipped ? style.hasExplicitlySetPaddingTop() : style.hasExplicitlySetPaddingBottom());
+
+    if (RefPtr input = dynamicDowncast<HTMLInputElement>(element); input && input->isNumberField() && !inlineEndPaddingExplicitlySet) {
+        const auto paddingBox = paddingBoxForNumberField(style, &element);
+
+        const auto paddingEdgeInlineStart = isHorizontal ? (isInlineFlipped ? paddingBox.right() : paddingBox.left()) : (isInlineFlipped ? paddingBox.bottom() : paddingBox.top());
+        const auto paddingEdgeInlineEnd = isHorizontal ? (isInlineFlipped ? paddingBox.left() : paddingBox.right()) : (isInlineFlipped ? paddingBox.top() : paddingBox.bottom());
```

Comment:
You should be able to use the logical variants, `start` and `end` here. See `RectEdges.h`.